src/Features/CSharp/Portable/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -91,6 +91,38 @@ internal static bool IsTriggerCharacter(SourceText text, int characterPosition,
             return false;
         }
 
+        /// <summary>
+        /// Tells if we are in positions like this: <c>#nullable $$</c> or <c>#pragma warning $$</c>
+        /// </summary>
+        internal static bool IsCompilerDirectiveTriggerCharacter(SourceText text, int characterPosition)
+        {
+            // First we check if current character is space
+            var ch = text[characterPosition--];
+            if (ch != ' ')
+            {
+                return false;
+            }
+
+            // Then step back while "current" character is letter
+            while (char.IsLetter(text[characterPosition]))
```

Comment:
I think we can step back for spaces (and possibly tabs) as well. Then this will avoid the recursion. If I'm missing something, let me know.